Add require-await lint rule

samrith-s opened the pull request:
<!--
Before submitting a PR, please read https://deno.land/std/manual.md#contributing
-->
Added `require-await` in linting rule to fix #4394.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/denoland/deno?pullRequest=4401) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno?pullRequest=4401) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno?pullRequest=4401) it.</sub>

nayeemrmn commented on cli/js/permissions.ts:
```diff
@@ -47,17 +47,17 @@ export class PermissionStatus {
-  async request(desc: PermissionDescriptor): Promise<PermissionStatus> {
+  request(desc: PermissionDescriptor): PermissionStatus {
     const state = permissionsOps.request(desc);
     return new PermissionStatus(state);
```
Wrap the return types here in `Promise.resolve()`.

samrith-s replied:
Gotcha. Thanks!

kitsonk commented:
@samrith-s so far so good!

bartlomieju commented:
@samrith-s thank you for this cumbersome work! I fixed some tests and reverted a few changes that weren't necessary.

samrith-s commented:
> @samrith-s thank you for this cumbersome work! I fixed some tests and reverted a few changes that weren't necessary. 

You’re welcome! I got stuck trying to fix these errors. Thank you so much for stepping in.
